tracing-subscriber/src/filter/targets.rs

```diff
@@ -0,0 +1,410 @@
+    /// enabled for spans and events that didn't match any of the provided targets.
+    ///
+    /// [`with_target`]: Targets::with_target
+    /// [`with_targets`]: Targets::with_targets
+    /// [`with_default`]: Targets::with_default
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Enables spans and events with [target]s starting with the provided target
+    /// prefix if they are at or below the provided [`LevelFilter`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tracing_subscriber::filter;
+    /// use tracing_core::Level;
+    ///
+    /// let filter = filter::Targets::new()
+    ///     // Enable the `INFO` level for anything in `my_crate`
+    ///     .with_target("my_crate", Level::INFO)
+    ///     // Enable the `DEBUG` level for a specific module.
+    ///     .with_target("my_crate::interesting_module", Level::DEBUG);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [`LevelFilter::OFF`] can be used to disable a particular target:
+    /// ```
+    /// use tracing_subscriber::filter::{Targets, LevelFilter};
+    /// use tracing_core::Level;
+    ///
+    /// let filter = Targets::new()
+    ///     .with_target("my_crate", Level::INFO)
+    ///     // Disable all traces from `annoying_module`.
+    ///     .with_target("my_crate::interesting_module", LevelFilter::OFF);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [target]: tracing_core::Metadata::target
+    pub fn with_target(mut self, target: impl ToString, level: impl Into<LevelFilter>) -> Self {
+        self.0.add(StaticDirective::new(
+            Some(target.to_string()),
+            Default::default(),
+            level.into(),
+        ));
+        self
+    }
+    /// Adds [target]s from an iterator of [target]-[`LevelFilter`] pairs to this filter.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tracing_subscriber::filter;
+    /// use tracing_core::Level;
+    ///
+    /// let filter = filter::Targets::new()
+    ///     .with_targets(vec![
+    ///         ("my_crate", Level::INFO),
+    ///         ("my_crate::some_module", Level::DEBUG),
+    ///         ("my_crate::other_module::cool_stuff", Level::TRACE),
+    ///         ("other_crate", Level::WARN)
+    ///     ]);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [`LevelFilter::OFF`] can be used to disable a particular target:
+    /// ```
+    /// use tracing_subscriber::filter::{Targets, LevelFilter};
+    /// use tracing_core::Level;
+    ///
+    /// let filter = Targets::new()
+    ///     .with_target("my_crate", Level::INFO)
+    ///     // Disable all traces from `annoying_module`.
+    ///     .with_target("my_crate::interesting_module", LevelFilter::OFF);
+    /// # drop(filter);
+    /// ```
+    ///
+    /// [target]: tracing_core::Metadata::target
+    pub fn with_targets<T, L>(mut self, targets: impl IntoIterator<Item = (T, L)>) -> Self
+    where
+        String: From<T>,
```

Comment:
I would expect `T: Into<String>` — there are strictly more Into impl, so into should be used in bounds.

Reply:
yeah, I changed all the other bounds to be `Into<String>` but must have missed this one, thanks for catching that!